parley/src/layout/mod.rs

```diff
@@ -181,23 +182,14 @@ impl<B: Brush> Layout<B> {
     /// You must perform line breaking prior to aligning, through [`Layout::break_lines`] or
     /// [`Layout::break_all_lines`]. If `container_width` is not specified, the layout's
     /// [`Layout::width`] is used.
-    ///
-    /// If `align_when_overflowing` is set to `true`, alignment will apply even if the line
-    /// contents are wider than the alignment width. If it is set to `false`, all overflowing lines
-    /// will be [`Alignment::Start`] aligned.
     pub fn align(
         &mut self,
         container_width: Option<f32>,
         alignment: Alignment,
-        align_when_overflowing: bool,
+        options: AlignmentOptions,
```

Comment:
I'm not convinced by this signature. Having a pile of arguments, and then an options struct doesn't quite seem right to me:
```rust
fn align(&mut self, alignment: Alignment, options: AlignmentOptions)
```
seems more natural, given that `container_width` is already optional.

Or even:
```rust
fn align(&mut self, options: AlignmentOptions)
```

Reply:
One issue with moving `container_width` into the options is that it cannot account for floated boxes yet, and I don't think we know what that's going to look like yet.

I certainly see the logic of moving `alignment` into `AlignmentOptions`.

Reply:
> Just another consideration: perhaps container_width shouldn't be optional. Aligning to the layout's calculated width is not necessarily a sensible default, as that width changes depending on how exactly lines were broken given the container width (max_advance) used for line breaking.

This could also make sense.